analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -294,6 +303,76 @@ class SpdxDocumentFile(
             }
         }
 
+    /**
+     * Maps a given identifier on an [SpdxPackage] or [SpdxExternalDocumentReference].
+     *
+     * @param doc [SpdxDocument] containing all [SpdxPackage] or [SpdxExternalDocumentReference] the id could refer to.
+     * @param source the identifier to be found in the [SpdxDocument].
+     */
+    private fun getSpdxPackageFromId(doc: SpdxDocument, source: String): SpdxPackage {
+        doc.packages.singleOrNull { it.spdxId == source }?.let { return it }
+
+        val externalDocumentReference = doc.externalDocumentRefs.singleOrNull {
+            val reducedIdWithoutPackageReference = source.split(":").first()
```

Comment:
I would prefer leaving it in the function, because the second part of the split string will be used to find the package in the referenced document in one of the later commits you suggested to be in a separate PR